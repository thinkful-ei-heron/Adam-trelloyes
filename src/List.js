import React from 'react';

import Card from './Card';
import './List.css';

List.defaultProps = {
    cards: []
}

function List (props) {
    return (
        <section className='List'>
            <header className='List-cards'>
                <h2>{props.header}</h2>
            </header>
            <div className='List-cards'>
                {props.cards.map(card => {
                    return (
                    <Card
                    key= {card.id}
                    title= {card.title}
                    content={card.content}
                    />
                )})}
            </div>
        </section>
    );
}

export default List;