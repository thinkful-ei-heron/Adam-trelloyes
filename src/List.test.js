import List from './List';
import { isTSAnyKeyword } from '@babel/types';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

describe('<List />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<List title='test title' content='test content' />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})

