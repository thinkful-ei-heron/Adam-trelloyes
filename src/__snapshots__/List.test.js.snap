// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-cards"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  />
</section>
`;
